app/models/manageiq/providers/google/resource.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Google
+  class Resource
+    def initialize(uri)
+      @type = self.class.parse_resource_type(uri)
+    end
+
+    attr_reader :type
```

Comment:
Well, I was considering making it public to be able to support unknown resources.  Most collections in the google api follow the pattern:

<baseurl>/<basepath>/<collection>/<name>

I could explicitly whitelist the ones I know about, but it feels like inevitably we'll have to come in here and update this class every time we want to use it on a new collection.

Thoughts?